packages/bvaughn-architecture-demo/components/lexical/plugins/code-completion/findMatches.ts

```diff
@@ -61,6 +61,20 @@ function fetchQueryData(
           !PREVIEW_CAN_OVERFLOW
         );
         properties = preview?.properties || null;
+
+        // Auto-complete should not just include own properties.
+        // If there's a prototype chain, fetch those properties also.
+        let currentPrototypeId = preview?.prototypeId;
+        while (currentPrototypeId) {
```

Comment:
:speech_balloon: only real question is if we'd want to limit the depth here

Reply:
Hm. Yeah, that's an interesting question. I wonder what that limit should be? I guess I could pick something arbitrary (like 10) to avoid an infinite loop but high enough to still probably handle 99% of the cases